src/core/qgshelp.cpp

```diff
@@ -0,0 +1,145 @@
+/***************************************************************************
+  qgshelp.cpp
+  --------------------------------------
+  Date                 : December 2016
+  Copyright            : (C) 2016 by Alexander Bruy
+  Email                : alexander dot bruy at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgshelp.h"
+
+#include <QSettings>
+#include <QLocale>
+#include <QUrl>
+#include <QFileInfo>
+#include <QTcpSocket>
+#include <QDesktopServices>
+
+#include "qgis.h"
+#include "qgsapplication.h"
+
+QgsHelp *QgsHelp::sHelp = nullptr; // Singleton instance
```

Comment:
We are avoiding adding new singletons to the code - see https://github.com/qgis/qgis3.0_api/issues/42

The correct approach here would be copying how the task manager/action scope registry/profiler are implemented in QgsApplication. See
https://github.com/qgis/QGIS/blob/master/src/core/qgsapplication.h#L371
https://github.com/qgis/QGIS/blob/master/src/core/qgsapplication.h#L510
https://github.com/qgis/QGIS/blob/master/src/core/qgsapplication.cpp#L112
https://github.com/qgis/QGIS/blob/master/src/core/qgsapplication.cpp#L247
https://github.com/qgis/QGIS/blob/master/src/core/qgsapplication.cpp#L1506